interp/interp.go

```diff
@@ -501,6 +500,7 @@ func (r *Runner) Reset() {
 		Exec:        r.Exec,
 		Open:        r.Open,
 		KillTimeout: r.KillTimeout,
+		opts:        r.opts,
```

Comment:
I thought of that but it *think* leaks shell options `set` in a script from one `Run` to another. Is this the intended behaviour?

Reply:
Yes, I've been thinking the same overnight. This is intended for now, because `Runner.opts` should be in the same camp as `Runner.Params`. They can both be set by the `Params` function, and by the `set` builtin.

Like you say, this isn't ideal, because `Reset` should leave constructor options intact, while undoing changes done by scripts. Still, it's better if both fields behave consistently for now; we can always refine the API to be saner in v3. That kind of change would likely be too invasive to backport to v2.